src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1536,6 +1536,168 @@ int SketchObject::delConstraintOnPoint(int GeoId, PointPos PosId, bool onlyCoinc
+                continue;
+            }
+
+            if (point1First || point2First) {
+                // Move the coincident constraint to the new corner point
+                c->First = originalCornerId;
+                c->FirstPos = start;
+            }
+
+            if (point1Second || point2Second) {
+                // Move the coincident constraint to the new corner point
+                c->Second = originalCornerId;
+                c->SecondPos = start;
+            }
+        } else if (c->Type == Sketcher::Distance || c->Type == Sketcher::DistanceX || c->Type == Sketcher::DistanceY) {
+            // Point-to-point distance constraint.  Move it to the new corner point
+            if (point1First || point2First) {
+                c->First = originalCornerId;
+                c->FirstPos = start;
+            }
+            if (point1Second || point2Second) {
+                c->Second = originalCornerId;
+                c->SecondPos = start;
+            }
+
+            // Distance constraint on the line itself. Change it to point-point between the far end of the line
+            // and the new corner
+            if (line1First) {
+                c->FirstPos = (posId1 == start) ? end : start;
+                c->Second = originalCornerId;
+                c->SecondPos = start;
+            }
+            if (line2First) {
+                c->FirstPos = (posId2 == start) ? end : start;
+                c->Second = originalCornerId;
+                c->SecondPos = start;
+            }
+        } else if (c->Type == Sketcher::PointOnObject) {
+            // The corner to be filleted was touching some other object.
+            if (point1First || point2First) {
+                c->First = originalCornerId;
+                c->FirstPos = start;
+            }
+        } else if (c->Type == Sketcher::Equal) {
+            // Equal length constraints are dicey because the lines are getting shorter.  Safer to
+            // delete them and let the user notice the underconstraint.
+            if (line1First || line2First || line1Second || line2Second) {
+                continue;
+            }
+        } else if (c->Type == Sketcher::Symmetric) {
+            // Symmetries should probably be preserved relative to the original corner
+            if (point1First || point2First) {
+                c->First = originalCornerId;
+                c->FirstPos = start;
+            } else if (point1Second || point2Second) {
+                c->Second = originalCornerId;
+                c->SecondPos = start;
+            } else if (point1Third || point2Third) {
+                c->Third = originalCornerId;
+                c->ThirdPos = start;
+            }
+        } else if (c->Type == Sketcher::SnellsLaw) {
+          // Can't imagine any cases where you'd fillet a vertex going through a lens, so let's
+          // delete to be safe.
+          continue;
+        }
+
+        // Default: copy this constraint to the new list

```

Comment:
By coding it this way you are implicitly keeping any constraint type defined in future versions: it might be safer to instead actually code the remainder of the conditional, with a final `else` clause that does _not_ keep the constraint.

Reply:
Great catch.  Fixed, and you made me realize I had missed point-to-point Horizontal and Vertical constraints.